rfd/0085-better-react-testing-remove-default-exports.md

```diff
@@ -0,0 +1,163 @@
+  return <DesktopSession {...state} />;
+}
+```
+
+
+### Testing
+
+As a result of the above, we end up testing the `DesktopSession` component instead of the actual component (`Container`) that is
+rendered in the UI.
+
+A story may look like this:
+
+```typescript jsx
+export const FetchError = () => (
+  <DesktopSession
+    {...props}
+    fetchAttempt={{ status: 'failed', statusText: 'some fetch  error' }}
+    tdpConnection={{ status: 'success' }}
+    wsConnection={'open'}
+    disconnected={false}
+  />
+);
+```
+
+Which is then tested like so:
+
+```typescript jsx
+test('fetch error', () => {
+  const { getByTestId } = render(<FetchError />);
+  expect(getByTestId('Modal')).toMatchSnapshot();
+});
+```
+
+This results in all the logic in `useDesktopSession` going untested. We're only testing the UI part of the application,
+but skipping all of the business logic.
+
+Storybook isn't best suited for the way we are using it. As a result of testing our large components primarily via stories, we have
+ended up only testing our UI and nothing else.
+
+## Details
+
+### Changing how we write stories
+
+Experience tells us that Stories in Storybook are used most effectively for documenting how to use reusable components. We should therefore
+restrict usage of Storybook mostly to our shared component library, and test our fullscreen components another way.
+
+It's still okay to create our larger components in Storybook (if the developer wants to) but we should not be using these stories in our Jest tests.
+
+It might be worthwhile looking into something like [Chromatic](https://www.chromatic.com/) for visual UI testing.
+
+### Changing how we write tests
+
+We should be testing all the different state possibilities in Jest and React Testing Library, mocking any data or
+network requests to cause the component to render into the state we want. This ensures that the internal logic
+of our components (generally the most critical and difficult-to-get-right aspect of frontend code) gets tested,
+as well as the final UI state.
+
+Tests should check for elements that exist for the specific state being tested, i.e. an error message when the data has
+failed to load. Elements should be clicked on or events triggered in order to change the state of the component.
+
+By writing tests as described above, snapshot tests typically become redundant and therefore unnecessary. Besides the problem of skipping the business logic, snapshots consistently break for non-bugs, such as when styles are updated, and
+generally don't provide much if any security around behaviour or appearance. Further reading -
+https://medium.com/@sapegin/whats-wrong-with-snapshot-tests-37fbe20dfe8e.
+
+### Changing how we export components
+
+Default exports should be avoided except when absolutely necessary (for example, with
+`React.Lazy`), as they degrade the developer experience
+for no discernible benefit. For more fully fleshed out reasoning, refer to the articles below:
```

Comment:
On top of saying what should be avoided unless absolutely necessary and giving an example of what to do when exporting something as default cannot be avoided, I think before mentioning those two facts we should show what practice we want to enforce, that is named exports.

That's just to make the RFD a little bit more clear in terms of guidance and explicitly say what we want to achieve vs saying just what we want to avoid.

Reply:
I've added an example of named exports and mentioned them